src/datasources/email/email.datasource.ts

```diff
@@ -199,4 +221,87 @@ export class EmailDataSource implements IEmailDataSource {
       );
     }
   }
+
+  async getSubscriptions(args: {
+    chainId: string;
+    safeAddress: string;
+    account: string;
+  }): Promise<DomainSubscription[]> {
+    const subscriptions = await this.sql`SELECT key, name
+                                         FROM emails.subscriptions
+                                                  INNER JOIN emails.account_subscriptions subs
+                                                             on subscriptions.id = subs.subscription_id
+                                                  INNER JOIN emails.account_emails emails on emails.id = subs.account_id
+                                         WHERE chain_id = ${args.chainId}
+                                           AND safe_address = ${args.safeAddress}
+                                           AND account = ${args.account}`;
+    return subscriptions.map(
+      (subscription) =>
+        <DomainSubscription>{
+          key: subscription.key,
+          name: subscription.name,
+        },
+    );
+  }
+
+  async subscribe(args: {
+    chainId: string;
+    safeAddress: string;
+    account: string;
+    categoryKey: string;
+  }): Promise<DomainSubscription[]> {
+    const subscriptions = await this
+      .sql`INSERT INTO emails.account_subscriptions (account_id, subscription_id)
+           SELECT ae.id AS account_id,
```

Comment:
Nit: as before regarding variable names. Please resolve if I misunderstand.

Reply:
Addressed in [040d7d6](https://github.com/safe-global/safe-client-gateway/pull/1028/commits/040d7d6378f9fe26daf364d342d3a9481c361b8a)